subsys/bluetooth/audio/has.c

```diff
@@ -1117,9 +1117,12 @@ int bt_has_preset_unregister(uint8_t index)
 	return control_point_send_all(&buf);
 }
 
-int bt_has_preset_available(uint8_t index)
+static int set_preset_availability(uint8_t index, bool available)
 {
+	NET_BUF_SIMPLE_DEFINE(buf, sizeof(struct bt_has_cp_hdr) +
+			      sizeof(struct bt_has_cp_preset_changed) + sizeof(uint8_t));
```

Comment:
> Which btw, this isn't really legal in C99?

IIRC Zephyr moved to C11 a while ago

Reply:
It is actually the same in ISO 9899:2011 6.7.6.2:
> If the expression is a constant expression, it shall have a value greater than zero.

However, according to https://docs.zephyrproject.org/latest/develop/languages/c/index.html Zephyr does not target a specific C standard, it just claims that many features introduced in C99 and C11 is used. Since they are all over the place in Zephyr it has to be fine. Just saying, I prefer the other solution :D

Reply:
You are right - The flexibly array (denoted by `[]` instead of `[0]`) is valid in C99 though, and the `[0]` seems to be a GCC extension (and also supported in LLVM). 

Nevertheless it's a common practice to use in Zephyr (searching for ` \w+\[0\];\n\}` yields 195 results, the majority of which are in the BT subsystem). It is probably something that should be replaced by `[]` or the newer bounded flexible arrays (https://people.kernel.org/kees/bounded-flexible-arrays-in-c)

Reply:
Created https://github.com/zephyrproject-rtos/zephyr/issues/63845

Reply:
Defining `sizeof(struct bt_has_cp_hdr) + sizeof(struct bt_has_cp_preset_changed) + sizeof(uint8_t)` as a single struct or value would still make sense to do in this PR IMO

Reply:
Defining `sizeof(struct bt_has_cp_hdr) + sizeof(struct bt_has_cp_preset_changed) + sizeof(uint8_t)` as a single struct or value would still make sense to do in this PR IMO

Reply:
Those structures are common for client and server. I would need to change the client code as well

Reply:
Not suggesting to change any existing, but rather a new one that suits this. Not sure I understand the issue with doing so